File: dotnet/Lexy/Lexy.Compiler/Compiler/CSharp/BuiltInFunctions/EndStartDateFunctionCall.cs
using Lexy.Compiler.Language.Expressions.Functions;
using Microsoft.CodeAnalysis;
using Microsoft.CodeAnalysis.CSharp;
using Microsoft.CodeAnalysis.CSharp.Syntax;

namespace Lexy.Compiler.Compiler.CSharp.BuiltInFunctions;

internal abstract class EndStartDateFunctionCall<TExpressionFunction> : MethodFunctionCall<TExpressionFunction>
    where TExpressionFunction : EndStartDateFunction
{
    protected override SeparatedSyntaxList<ArgumentSyntax> GetArguments(TExpressionFunction powerFunction)
    {
        return SyntaxFactory.SeparatedList<ArgumentSyntax>(
            new SyntaxNodeOrToken[]
            {
                SyntaxFactory.Argument(ExpressionSyntaxFactory.ExpressionSyntax(powerFunction.EndDateExpression)),
                SyntaxFactory.Token(SyntaxKind.CommaToken),
                SyntaxFactory.Argument(ExpressionSyntaxFactory.ExpressionSyntax(powerFunction.StartDateExpression))
            });
    }
}

File: typescipt/existing/src/compiler/cSharp/builtInFunctions/endStartDateFunctionCall.ts





namespace Lexy.Compiler.Compiler.CSharp.BuiltInFunctions;

internal abstract class EndStartDateFunctionCall : MethodFunctionCall
{
   public EndStartDateFunction Function { get; }

   protected EndStartDateFunctionCall(EndStartDateFunction function) : base(function)
   {
     Function = function;
   }

   public override MemberDeclarationSyntax CustomMethodSyntax(ICompileFunctionContext context)
   {
     return null;
   }

   protected override SeparatedSyntaxList<ArgumentSyntax> GetArguments(ICompileFunctionContext context)
   {
     return SyntaxFactory.SeparatedList<ArgumentSyntax>(
       new SyntaxNodeOrToken[]
       {
         SyntaxFactory.Argument(ExpressionSyntaxFactory.ExpressionSyntax(Function.EndDateExpression, context)),
         SyntaxFactory.Token(SyntaxKind.CommaToken),
         SyntaxFactory.Argument(ExpressionSyntaxFactory.ExpressionSyntax(Function.StartDateExpression, context))
       });
   }
}
